controllers/addresspool_controller.go

```diff
@@ -69,15 +69,18 @@ func (r *AddressPoolReconciler) ReconcileResource(client *gophercloud.ServiceCli
 				return common.NewUserDataError(msg)
 			}
 
-			host_instance.Status.InSync = false
-			host_instance.Status.Reconciled = false
-			err = r.Client.Status().Update(context.TODO(), host_instance)
+			err = r.CloudManager.NotifyResource(host_instance)
```

Comment:
Why to update the existing method here? Expect a comment to explain it. And please update the method description above

Reply:
Sometimes I've noticed that when the `reconciled` and `insync` are already set to false it won't trigger reconciliation even though address pool / platformnetwork are updated.
To circumvent this problem, we are re-using the functionality of updating "deployment-manager/notification" annotation in combination with `PlatformNetworkUpdateRequired` function which determines whether platformnetwork sub-reconciliation should be executed.

I'll update the method description as per the new behavior.